src/main/java/com/redhat/coolstore/service/ShoppingCartOrderProcessor.java

```diff
@@ -1,35 +1,32 @@
 package com.redhat.coolstore.service;
 
 import java.util.logging.Logger;
-import javax.ejb.Stateless;
-import javax.annotation.Resource;
 import javax.inject.Inject;
-import javax.jms.JMSContext;
-import javax.jms.Topic;
+import io.quarkus.vertx.ConsumeEvent;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.reactive.messaging.annotations.Channel;
+import io.smallrye.reactive.messaging.annotations.Emitter;
+import io.smallrye.reactive.messaging.Message; // Updated import for SmallRye Reactive Messaging Message
+import javax.enterprise.context.ApplicationScoped; // Added import for CDI ApplicationScoped
 
 import com.redhat.coolstore.model.ShoppingCart;
 import com.redhat.coolstore.utils.Transformers;
 
-@Stateless
-public class ShoppingCartOrderProcessor  {
+@ApplicationScoped // Replaced @Stateless with @ApplicationScoped for CDI bean
+public class ShoppingCartOrderProcessor {
 
     @Inject
     Logger log;
 
-
     @Inject
-    private transient JMSContext context;
+    private transient Message<String> message; // Replaced JMSContext with Message
 
-    @Resource(lookup = "java:/topic/orders")
-    private Topic ordersTopic;
+    @Inject
+    @Channel("orders")
+    Emitter<String> ordersEmitter;
```

Comment:
Here @christophermay07 added a `@Broadcast` annotation. I don't think we have a rule for that